### ISSUE TITLE
questions on validation set and threshold selection algorithm

### ISSUE THREAD

**Issue by carrtesy** [thuml/Anomaly-Transformer#32]
Hello,

I have a question regarding your work.

1. How validation set are selected here? from [dataloader code](https://github.com/thuml/Anomaly-Transformer/blob/72a71e5f0847bd14ba0253de899f7b0d5ba6ee97/data_factory/data_loader.py#L37), validation set seems to be test set.  Did I get the codes right?
2.  Anomaly Transformer's thresholding mechanism.
Is the model using test dataset rather than validation set to set the threshold?
From Appendix H, validation set (which is consisted of normal data only) is used to pick an appropriate threshold.
However, [the code](https://github.com/thuml/Anomaly-Transformer/blob/72a71e5f0847bd14ba0253de899f7b0d5ba6ee97/solver.py#L287) seems to utilize test dataset for thresholding.   
Did I miss some points?

Thanks a lot for your answer in advance.

**Comment by xjw-wade** [comment]
I have the same question too, thanks a lot for your answer in advance.

**Comment by carrtesy** [comment]
This issue seems to be addressed on [previous issue](https://github.com/thuml/Anomaly-Transformer/issues/27).
However, I still wonder:
(1) If the F1-score reported on the paper use validation set to attain threshold (Implementation details of Section 4. Experiement states that the thredhold $\delta$ is determined using validation dataset.) 
(2) If the Protocol of threshold selection (appendix H) are guaranteed to provide threshold that are close to the optimal threshold.
